include/ddsrouter/participant/ParticipantDatabase.hpp

```diff
@@ -50,6 +52,8 @@ class ParticipantDatabase
 protected:
 
     std::map<ParticipantId, std::shared_ptr<IParticipant>> participants_;
+
+    mutable std::shared_timed_mutex mutex_;
```

Comment:
Would perhaps a shared_mutex be enough?

Reply:
`shared_mutex` is not added until C++17
https://en.cppreference.com/w/cpp/thread/shared_mutex